src/client/components/GetMeal.js

```diff
@@ -0,0 +1,129 @@
+import AddReservation from "./AddReservation";
+import React, { useState, useEffect } from "react";
+import { useParams} from 'react-router-dom';
+import MealsImages from "./MealsImages";
+
+const GetMeal = ()=>{
```

Comment:
You need more separation here. It's usually easy tell if a component does more than one thing. Here you can see it just by looking at the name. GetMeal suggests getting a specific meal from the backend but the component does much than that. 

Think about how your app might change in the future. Could there be more than one reason for this component to change?